Pywebsocket refactoring

arenevier opened the pull request.

arenevier commented:
@ricea

ricea commented on tools/wptserve/wptserve/ws_h2_handshake.py:
```diff
@@ -8,39 +8,21 @@
+class WsH2Handshaker(HandshakerBase):
     def __init__(self, request, dispatcher):
-        """Opening handshake processor for the WebSocket protocol (RFC 6455).
+        """Bootstraping handshake processor for the WebSocket protocol with HTTP/2 (RFC 8441).
```
"Bootstrapping"

ricea reviewed (commented):
lgtm

I verified the `tools/third_party/pywebsocket3` files are identical to upstream commit [265579a969e1aec2a6457db7674a9e31fbb7d564 ](https://github.com/GoogleChromeLabs/pywebsocket3/commit/265579a969e1aec2a6457db7674a9e31fbb7d564).

arenevier commented:
Are the tests failure on windows legitimate failures?
https://dev.azure.com/web-platform-tests/b14026b4-9423-4454-858f-bf76cf6d1faa/_apis/build/builds/64487/logs/203

I don't have a windows machine. But on linux, I can't reproduce when running
$ tox -c tools/wpt/ -e py38 -- --junitxml=results.xml

Commit:
update ws_h2_handshake.py to recent pywebsocket

pywebsocket has been refactored to expose some utilities and base class
as part of its public API.
(see https://github.com/GoogleChromeLabs/pywebsocket3/pull/28)

By updating to the newest pywebsocket3 version, we can make
ws_h2_handshake much simpler

LukeZielinski reviewed (approved):
Check failures are unrelated (#28745)

foolip commented:
@arenevier can you try rebasing? Those CI failures should be resolved now.

Also, everyone on this PR might be interested in https://github.com/web-platform-tests/wpt/issues/28863, which I filed after noticing we're going in an amusing direction :)

Commit:
Squashed 'tools/third_party/pywebsocket3/' changes from 9db3d29db4..265579a969

265579a969 Expose base.py as a public api. (#28)
cdb83e3152 Add missing r to regular expression (#27)

git-subtree-dir: tools/third_party/pywebsocket3
git-subtree-split: 265579a969e1aec2a6457db7674a9e31fbb7d564

Commit:
Merge commit '2a247ac52a4af9f82d50b2f96a43abe5c4629ed3' into HEAD

arenevier commented:
> @arenevier can you try rebasing? Those CI failures should be resolved now.


Done.

I have separeted in two commits: first the merge obtained with

```git subtree pull --prefix tools/third_party/pywebsocket3 https://github.com/GoogleChromeLabs/pywebsocket3 master `--squash```

Then, the actual changes to wpt.

I'm not sure if it would be better to squash them.

foolip commented:
In order to preserve the history correctly and make future `git subtree` calls work correctly it will be necessary to merge this into wpt with branch protection temporarily disabled.

Right @stephenmcgruer, is that how you've been doing it?

ricea commented:
Ping?

foolip commented:
@jgraham let me know on IRC that the branch protection rules don't apply to admins, which I can confirm in settings. Makes sense, that's the same reason why admin merging in the UI is allowed. (Force pushing or deleting `master` is still not allowed for admins, fortunately.)

I'll go ahead and merge this from the command line.
